ci/codecov.sh

```diff
@@ -55,19 +55,27 @@ elif [[ "$1" == "upload" ]]; then
     : "${LCOV_BRANCH_COVERAGE:=1}" # Set default
 
     # coverage files are in ../../b2 from this location
-    lcov --gcov-tool=$GCOV --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE} --base-directory "$BOOST_ROOT/libs/$SELF" --directory "$BOOST_ROOT" --capture --output-file all.info
+    lcov --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE} --gcov-tool=${GCOV} --directory "${BOOST_ROOT}" --capture --output-file all.info
     # dump a summary on the console
-    lcov --list all.info
+    lcov --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE} --list all.info
 
     # all.info contains all the coverage info for all projects - limit to ours
     # first we extract the interesting headers for our project then we use that list to extract the right things
     for f in `for f in include/boost/*; do echo $f; done | cut -f2- -d/`; do echo "*/$f*"; done > /tmp/interesting
     echo headers that matter:
     cat /tmp/interesting
-    xargs -L 999999 -a /tmp/interesting lcov --gcov-tool=$GCOV --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE:-1} --extract all.info {} "*/libs/$SELF/*" --output-file coverage.info
+    xargs -L 999999 -a /tmp/interesting lcov --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE} --extract all.info {} "*/libs/$SELF/*" --output-file coverage.info
+    
+    if [ ${LCOV_REMOVE_TEST_DIR} == 1 ]; then
+        lcov --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE} --list coverage.info
+        mv coverage.info coverage-with-test.info
+        lcov --rc lcov_branch_coverage=${LCOV_BRANCH_COVERAGE} --remove coverage-with-test.info "*/libs/$SELF/test/*" --output-file coverage.info
+    fi
 
     # dump a summary on the console - helps us identify problems in pathing
-    lcov --list coverage.info
+    # note this has test file coverage in it - if you do not want to count test
+    # files against your coverage numbers (who does?) then use a .codecov.yml file
```

Comment:
The comment is slightly wrong: The file won't have test coverage if LCOV_REMOVE_TEST_DIR is set.

I'd remove the variable but include a note that the .codecov.yml file needs to be in the default branch of the repo to be considered. (I admit I missed that too)

I'm not fully against LCOV_REMOVE_TEST_DIR  but I'd rather not have it as otherwise there are 2 methods to do essentially the same thing.

Reply:
I did not realize it did not pull .codecov.yml from the current branch.  Okay that explains why I could not get it to work at all.  Thank you.